tests/python_functional/src/driver_io/network/network_io.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2015-2018 Balabit
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+from pathlib2 import Path
+
+import src.testcase_parameters.testcase_parameters as tc_parameters
+from src.common.random_id import get_unique_id
+from src.driver_io.file.file import File
+from src.helpers.loggen.loggen import Loggen
+
+
+class NetworkIO():
+    def __init__(self, ip, port, transport):
+        self.__ip = ip
+        self.__port = port
+        self.__transport = transport.replace("_", "-").replace("'", "").replace('"', "")
+
+    def write(self, content):
+        loggen_input_file_path = Path(tc_parameters.WORKING_DIR, "loggen_input_{}.txt".format(get_unique_id()))
+        loggen_input_file = File(loggen_input_file_path)
+        with loggen_input_file.open_file(mode="a+") as f:
+            f.write(content)
+
+        transport_params = self.__transport_to_params(self.__transport)
+        Loggen().start(self.__ip, self.__port, read_file=str(loggen_input_file_path), dont_parse=True, permanent=True, **transport_params)
+
+    @staticmethod
+    def __transport_to_params(transport):
```

Comment:
@furiel What would be the benefit of it?

Reply:
It improves readability: `NetworkIO` class can be shorter. We also have less code with indentation.
It makes a little harder to work with repl: it is harder to copy it into the repl if you want to play around with it, because you need to pull the entire `NetworkIO`. If you want to update it's behavior in the repl, you need to overwrite the entire class.
Similarly, if you want to unit test it, you cannot call it without `NetworkIO`.
Unintentionally strengthens the justification for `NetworkIO`, just because that class stores its code.
It causes a little confusion, because it actually couples with `write` and not the entire class. Technically, this should be a local function inside `write`. But as this function longer than it would conveniently (readably) fit into `write`.

In general, `@staticmethod` is impractical for information hiding or namespacing. On the other hand, if you meta-program a class, it is a good choice. For example

```py
class Loggen(object):

    instanceIndex = -1
    @staticmethod
    def __get_new_instance_index():
        Loggen.instanceIndex += 1
        return Loggen.instanceIndex
```
is a justifiable application for `@staticmethod`.

Reply:
Removed this function all together.